scripts/gpgkey/main.go

```diff
@@ -0,0 +1,82 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package main
+
+import (
+	"log"
+	"os"
+	"strings"
+
+	"github.com/ProtonMail/gopenpgp/v2/crypto"
+)
+
+const (
+	signerPassphraseEnv = "ELASTIC_PACKAGE_SIGNER_PASSPHRASE"
+	privateKeyPathEnv   = "ELASTIC_PACKAGE_SIGNER_PRIVATE_KEYFILE"
+)
+
+func main() {
+	passphrase := readPassphrase()
+	privateKeyPath, publicKeyPath := getKeyPaths()
+	rsaKeyArmor, rsaPublicKeyArmor := genKey(passphrase)
+
+	err := os.WriteFile(privateKeyPath, rsaKeyArmor, 0644)
+	if err != nil {
+		log.Fatal(err)
+	}
+	err = os.WriteFile(publicKeyPath, rsaPublicKeyArmor, 0644)
+	if err != nil {
+		log.Fatal(err)
+	}
+}
+
+func genKey(passphrase []byte) ([]byte, []byte) {
+	const (
+		name    = "Elastic Package Test"
+		rsaBits = 2048
+	)
+
+	rsaKey, err := crypto.GenerateKey(name, "", "rsa", rsaBits)
+	if err != nil {
+		log.Fatal(err)
+	}
+	rsaKey, err = rsaKey.Lock(passphrase)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	rsaKeyArmor, err := rsaKey.Armor()
+	if err != nil {
+		log.Fatal(err)
+	}
+	rsaPublicKeyArmor, err := rsaKey.GetArmoredPublicKey()
```

Comment:
Not familiar with this library,
should this be using `rsaKeyArmor` instead of `rsaKey`?

```suggestion
	rsaPublicKeyArmor, err := rsaKeyArmor.GetArmoredPublicKey()
```

Reply:
I don't think so. `rsaKey` contains information about both the private key and its public key. "Armoring" here only refers to getting each one of the keys in PEM format, the format that elastic-package expects to find.

Reply:
`rsaKeyArmor` is a string.

Reply:
Ah Ok!
Thanks!